internal/driver/onvifdiscovery.go

```diff
@@ -0,0 +1,235 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"fmt"
+	"github.com/IOTechSystems/onvif"
+	wsdiscovery "github.com/IOTechSystems/onvif/ws-discovery"
+	"github.com/edgexfoundry/device-onvif-camera/pkg/netscan"
+	sdkModel "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+	"github.com/gofrs/uuid"
+	"github.com/pkg/errors"
+	"net"
+	"os"
+	"strings"
+	"time"
+)
+
+const (
+	bufSize = 8192
+)
+
+// OnvifProtocolDiscovery implements netscan.ProtocolSpecificDiscovery
+type OnvifProtocolDiscovery struct {
+	driver *Driver
+}
+
+func NewOnvifProtocolDiscovery(driver *Driver) *OnvifProtocolDiscovery {
+	return &OnvifProtocolDiscovery{driver: driver}
+}
+
+// ProbeFilter takes in a host and a slice of ports to be scanned. It should return a slice
+// of ports to actually scan, or a nil/empty slice if the host is to not be scanned at all.
+// Can be used to filter out known devices from being probed again if required.
+func (proto *OnvifProtocolDiscovery) ProbeFilter(_ string, ports []string) []string {
+	// For onvif we do not want to do any filtering
+	return ports
+}
+
+// OnConnectionDialed handles the protocol specific verification if there is actually
+// a valid device or devices at the other end of the connection.
+func (proto *OnvifProtocolDiscovery) OnConnectionDialed(host string, port string, conn net.Conn, params netscan.Params) ([]netscan.ProbeResult, error) {
+	// attempt a basic direct probe approach using the open connection
+	devices, err := executeRawProbe(conn, params)
+	if err != nil {
+		params.Logger.Debugf(err.Error())
```

Comment:
Comment why using Debug and not Error or Warn
```suggestion
		params.Logger.Debug(err.Error())
```